### ISSUE TITLE
npm don't work in Mac

### ISSUE THREAD

**Issue by farnetani** [start-angular/SB-Admin-BS4-Angular-2#38]
I did installation of node LTS 4.4.4, but don't work.

sh-3.2# cat npm-debug.log
0 info it worked if it ends with ok
1 verbose cli [ '/usr/local/bin/node', '/usr/local/bin/npm', 'start' ]
2 info using npm@2.15.1
3 info using node@v4.4.4
4 verbose run-script [ 'prestart', 'start', 'poststart' ]
5 info prestart SB-Admin-BS4-Angular-2@0.0.0
6 info start SB-Admin-BS4-Angular-2@0.0.0
7 verbose unsafe-perm in lifecycle true
8 info SB-Admin-BS4-Angular-2@0.0.0 Failed to exec start script
9 verbose stack Error: SB-Admin-BS4-Angular-2@0.0.0 start: `gulp serve.dev --color`
9 verbose stack Exit status 1
9 verbose stack     at EventEmitter.<anonymous> (/usr/local/lib/node_modules/npm/lib/utils/lifecycle.js:217:16)
9 verbose stack     at emitTwo (events.js:87:13)
9 verbose stack     at EventEmitter.emit (events.js:172:7)
9 verbose stack     at ChildProcess.<anonymous> (/usr/local/lib/node_modules/npm/lib/utils/spawn.js:24:14)
9 verbose stack     at emitTwo (events.js:87:13)
9 verbose stack     at ChildProcess.emit (events.js:172:7)
9 verbose stack     at maybeClose (internal/child_process.js:827:16)
9 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
10 verbose pkgid SB-Admin-BS4-Angular-2@0.0.0
11 verbose cwd /Volumes/WINDOWS/Dropbox (Pessoal)/Dropbox - 2016/www-2016/Angular2/SB-Admin-BS4-Angular-2
12 error Darwin 15.5.0
13 error argv "/usr/local/bin/node" "/usr/local/bin/npm" "start"
14 error node v4.4.4
15 error npm  v2.15.1
16 error code ELIFECYCLE
17 error SB-Admin-BS4-Angular-2@0.0.0 start: `gulp serve.dev --color`
17 error Exit status 1
18 error Failed at the SB-Admin-BS4-Angular-2@0.0.0 start script 'gulp serve.dev --color'.
18 error This is most likely a problem with the SB-Admin-BS4-Angular-2 package,
18 error not with npm itself.
18 error Tell the author that this fails on your system:
18 error     gulp serve.dev --color
18 error You can get information on how to open an issue for this project with:
18 error     npm bugs SB-Admin-BS4-Angular-2
18 error Or if that isn't available, you can get their info via:
18 error
18 error     npm owner ls SB-Admin-BS4-Angular-2
18 error There is likely additional logging output above.
19 verbose exit [ 1, true ]
sh-3.2# https://nodejs.org/dist/v4.4.4/node-v4.4.4.pkg

**Comment by nitinkumarmishra** [comment]
please provide us your system info like node version , npm version & gulp etc

**Comment by jamesd-udley** [comment]
Getting the same error on Windows 10 x64

System info:
````
OS: Windows 10 x64 (14393)
npm version: 3.10.3
node version: 6.6.0
gulp version: 3.9.1
`````

debug.log:
````
0 info it worked if it ends with ok
1 verbose cli [ 'C:\\Program Files\\nodejs\\node.exe',
1 verbose cli   'C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js',
1 verbose cli   'start' ]
2 info using npm@3.10.3
3 info using node@v6.6.0
4 verbose run-script [ 'prestart', 'start', 'poststart' ]
5 info lifecycle SB-Admin-BS4-Angular-2@0.0.0~prestart: SB-Admin-BS4-Angular-2@0.0.0
6 silly lifecycle SB-Admin-BS4-Angular-2@0.0.0~prestart: no script for prestart, continuing
7 info lifecycle SB-Admin-BS4-Angular-2@0.0.0~start: SB-Admin-BS4-Angular-2@0.0.0
8 verbose lifecycle SB-Admin-BS4-Angular-2@0.0.0~start: unsafe-perm in lifecycle true
9 verbose lifecycle SB-Admin-BS4-Angular-2@0.0.0~start: PATH: C:\Program Files\nodejs\node_modules\npm\bin\node-gyp-bin;C:\Users\----\Desktop\SB-Admin-BS4-Angular-2\node_modules\.bin;C:\Program Files\nodejs;C:\Python27\;C:\Python27\Scripts;C:\Program Files\Docker\Docker\Resources\bin;C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Windows Azure Tools\Docker\;C:\ProgramData\Oracle\Java\javapath;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Program Files\Microsoft SQL Server\120\Tools\Binn\;C:\Program Files\Microsoft SQL Server\110\Tools\Binn\;C:\Program Files\Microsoft\Web Platform Installer\;C:\Program Files (x86)\Microsoft SQL Server\130\Tools\Binn\ManagementStudio\;C:\Program Files (x86)\Microsoft SQL Server\130\Tools\Binn\;C:\Program Files (x86)\Microsoft SQL Server\Client SDK\ODBC\110\Tools\Binn\;C:\Program Files (x86)\Microsoft SQL Server\130\DTS\Binn\;C:\Program Files (x86)\Bitvise SSH Client;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\Program Files (x86)\Skype\Phone\;C:\Program Files\Microsoft SQL Server\130\Tools\Binn\;C:\Program Files (x86)\Microsoft SQL Server\Client SDK\ODBC\130\Tools\Binn\;C:\Program Files (x86)\Microsoft SQL Server\110\DTS\Binn\;C:\Program Files (x86)\Microsoft SQL Server\120\DTS\Binn\;C:\Program Files (x86)\Windows Kits\10\Windows Performance Toolkit\;C:\ProgramData\chocolatey\bin;C:\Program Files\Git\cmd;C:\Program Files\dotnet\;C:\Program Files\nodejs\;C:\Users\----\.dnx\runtimes\dnx-clr-win-x86.1.0.0-rc1-update1\bin;C:\Users\----\.dnx\bin;C:\Program Files (x86)\Microsoft VS Code\bin;C:\Users\----\AppData\Local\Microsoft\WindowsApps;C:\Users\----\AppData\Roaming\npm
10 verbose lifecycle SB-Admin-BS4-Angular-2@0.0.0~start: CWD: C:\Users\----\Desktop\SB-Admin-BS4-Angular-2
11 silly lifecycle SB-Admin-BS4-Angular-2@0.0.0~start: Args: [ '/d /s /c', 'gulp serve.dev --color' ]
12 silly lifecycle SB-Admin-BS4-Angular-2@0.0.0~start: Returned: code: 1  signal: null
13 info lifecycle SB-Admin-BS4-Angular-2@0.0.0~start: Failed to exec start script
14 verbose stack Error: SB-Admin-BS4-Angular-2@0.0.0 start: `gulp serve.dev --color`
14 verbose stack Exit status 1
14 verbose stack     at EventEmitter.<anonymous> (C:\Program Files\nodejs\node_modules\npm\lib\utils\lifecycle.js:242:16)
14 verbose stack     at emitTwo (events.js:106:13)
14 verbose stack     at EventEmitter.emit (events.js:191:7)
14 verbose stack     at ChildProcess.<anonymous> (C:\Program Files\nodejs\node_modules\npm\lib\utils\spawn.js:40:14)
14 verbose stack     at emitTwo (events.js:106:13)
14 verbose stack     at ChildProcess.emit (events.js:191:7)
14 verbose stack     at maybeClose (internal/child_process.js:877:16)
14 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:226:5)
15 verbose pkgid SB-Admin-BS4-Angular-2@0.0.0
16 verbose cwd C:\Users\-----\Desktop\SB-Admin-BS4-Angular-2
17 error Windows_NT 10.0.14393
18 error argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "start"
19 error node v6.6.0
20 error npm  v3.10.3
21 error code ELIFECYCLE
22 error SB-Admin-BS4-Angular-2@0.0.0 start: `gulp serve.dev --color`
22 error Exit status 1
23 error Failed at the SB-Admin-BS4-Angular-2@0.0.0 start script 'gulp serve.dev --color'.
23 error Make sure you have the latest version of node.js and npm installed.
23 error If you do, this is most likely a problem with the SB-Admin-BS4-Angular-2 package,
23 error not with npm itself.
23 error Tell the author that this fails on your system:
23 error     gulp serve.dev --color
23 error You can get information on how to open an issue for this project with:
23 error     npm bugs SB-Admin-BS4-Angular-2
23 error Or if that isn't available, you can get their info via:
23 error     npm owner ls SB-Admin-BS4-Angular-2
23 error There is likely additional logging output above.
24 verbose exit [ 1, true ]
````

**Comment by jamesd-udley** [comment]
**The problem seems to occur when there are ts-lint errors in the code base and `npm install` or `npm update` is run on the local packages.**

@farnetani - As a temporary solution make sure all of your ts-lint errors are resolved.

Cloned and built a new project didn't have this issue - must have to do with something that was modified inside the project.

Attempt 1: Tried deleting node_modules, ran `npm cache clean` and ran `npm install`.  Problem persists.

Attempt 2: Verified package.json matched this repo; re-did steps from Attempt 1 - same problem.

Attempt 3: Noticed during `npm start` there were a few ts-lint errors.  Resolved the ts-lint errors and re-did steps from Attempt 1.  Problem was resolved.  These were linting errors I had in a module I had created; not anything that shipped with this repository.
